lib/api/calendars.ts

```diff
@@ -51,3 +51,17 @@ export const modifyEvent = (client: DAVClient) => async (eventPath: string, even
     body: translate(event)
   });
 }
+
+export const changeParticipation = (client: DAVClient) => async (eventPath: string, mailAttendee: string, action: string, event: CalendarEventObject): Promise<string> => {
+  if (event.attendees) {
+    event.attendees.map(attendee => {
+      if (attendee.email == mailAttendee) {
+        attendee.partstat = action;
+       
+        return  modifyEvent(client)(eventPath, event);
+      }
+    })
+  }
+
+  return Promise.reject('can not change participation')
```

Comment:
```suggestion
  if (!Array.isArray(event.attendees)) return Promise.reject('Can not change participation');

  for (const attendee of event.attendees) {
    if (attendee.email !== attendeeEmail) continue;
    
    attendee.partstat = action;
    
    return modifyEvent(client)(eventPath, event);
  }

  return Promise.reject('No matching attendee found in the event')
```

Reply:
You should avoid nesting `if`s whenever possible. The code will be much more readable without nested `if`s.